### DISCUSSION TITLE
Résoudre facilement l'erreur 503 sous Kubernetes

### DISCUSSION THREAD

**Discussion by giscus** [zero-filtre/zerofiltre-blog#238] [Announcements]
# Résoudre facilement l'erreur 503 sous Kubernetes

Que faire lorsque votre application déployée sous k8s vous renvoit ce fameux 503 ?

https://zerofiltre.tech/articles/280-r%C3%A9soudre-facilement-l-erreur-503-sous-kubernetes

**Comment by willbrid** [comment]
Salut. De prime abord magnifique article. 

Je souhaiterais aussi apporter une petite contribution en disant ceci : ce cas de 503 peut aussi arriver lorsqu'on a confondu de port d'écoute du conteneur. Par exemple le conteneur écoute sur le port 80 mais l'on a mis sur l'option containerPort du déploiement ou statefulset le port 8080 alors on pourra faire :
- si le conteneur posséde les commandes curl ou wget , on pourra exécuter :

```
curl http://localhost:8080
```

ou

```
wget http://localhost:8080
```

- si le conteneur ne possède pas ces commandes, alors l'on pourra déployer le conteneur netshoot (nicolaka/netshoot) qui possède ces commandes afin d'effectuer le test via le nom du service de l'application

```
curl http://<nom_du_service.nom_du_namespace.svc>:8080
```

ou

```
wget http://<nom_du_service.nom_du_namespace.svc>:8080
```

**Reply by imphilippesimo** [reply]
Merci pour ton apport pertinent. +1 pour l'astuce en cas d'absence de shell dans le container.
